apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -327,19 +327,22 @@ resume_from_blocked(Data) ->
                     {next_state, running, Data}
             end;
         {expired, Ref, Batch} ->
-            IsAcked = ack_inflight(InflightTID, Ref, Id, Index),
-            IsAcked andalso emqx_resource_metrics:dropped_expired_inc(Id, length(Batch)),
+            %% assert true
+            true = ack_inflight(InflightTID, Ref, Id, Index),
```

Comment:
Q: if it's inflight, it might also get ack'ed from the async return fn as a late reply?

Reply:
true, I forgot about this race condition.